### ISSUE TITLE
Wrong value assigned to primary key model attribute (sqlserver)

### ISSUE THREAD

**Issue by daliran** [yiisoft/yii2#9401]
If the primary key of a sqlserver table has a function as default value, for example:

``` SQL
CREATE TABLE [dbo].[TestTable](
[Id] [uniqueidentifier] NOT NULL CONSTRAINT [DF_TestTable_Id]  DEFAULT (newid())
CONSTRAINT [id] PRIMARY KEY CLUSTERED )
```

the insert works fine on the database (I get the uuid correctly generated), but the model attribute has a wrong value.

``` php
$record = new TestTable();
$record->save();
var_dump($record->Id); // string(9) "(newid())"
```

The problem seems to be in the yii\db\Schema class in the insert() method: the plain default value is assigned to the model primary key attribute (without evaluating the expression), if the primary key is not autoincrement.

``` php
// yii\db\Schema (lines 454-459)
if ($tableSchema->columns[$name]->autoIncrement) {
    $result[$name] = $this->getLastInsertID($tableSchema->sequenceName);
    break;
} else {
    $result[$name] = isset($columns[$name]) ? $columns[$name] : $tableSchema->columns[$name]->defaultValue;
}
```

**Comment by cebe** [comment]
is there a way to get this value from the SQL server? Postgres allows us to fetch the value using a `RETURNING` statement: https://github.com/yiisoft/yii2/blob/master/framework/db/pgsql/Schema.php#L468

**Comment by pogush** [comment]
OUTPUT clause could be used for that: https://msdn.microsoft.com/en-us/library/ms177564.aspx

Example:
```
INSERT INTO [dbo].[test]
	([testField])
	OUTPUT inserted.[id]
VALUES
	('testValue') 
```

**Comment by nineinchnick** [comment]
The `yii\db\mssql\Schema.insert()` method needs to be overriden. I did not implement this when #8071 was being accepted because I don't have access to a MSSQL server.

**Comment by darkdef** [comment]
It's fixed in pr #18101
